app/scripts/components/common/mapbox/layers/raster-timeseries.tsx

```diff
@@ -206,16 +207,16 @@ export function MapLayerRasterTimeseries(props: MapLayerRasterTimeseriesProps) {
     };
     load();
     return () => {
-      controller && controller.abort();
-      changeStatus?.({ status: 'idle', context: STATUS_KEY.StacSearch });
+      controller.abort();
+      changeStatus({ status: 'idle', context: STATUS_KEY.StacSearch });
     };
   }, [id, changeStatus, stacCol, date]);
 
   //
   // Markers
   //
   useEffect(() => {
-    if (!id || !stacCol || !date || !minZoom || !stacCollection) return;
+    if (!id || !stacCol || !date || !minZoom || !stacCollection.length) return;
```

Comment:
This was an actual bug since `stacCollection` is always defined, just an empty array

Reply:
Nice catch!

Reply:
Careful with `minZoom` - `-Infinity` is truthy.